### ISSUE TITLE
Move to_pandas/from_pandas improvements upstream

### ISSUE THREAD

**Issue by astrofrog** [aperiosoftware/astropy-timeseries#14]
Rather than overloading to_pandas/from_pandas here, we should be able to implement support for handling ``Time`` columns in the core package directly.

**Comment by astrofrog** [comment]
This is done in https://github.com/astropy/astropy/pull/8247 - once that is merged, we should update the implementation here (or remove it if it's no longer needed, although if possible we should maybe check for the astropy version and for <3.2 we should keep the current implementation)
